crates/sui-types/src/messages.rs

```diff
@@ -1586,10 +1609,10 @@ impl TransactionData {
         self.kind.shared_input_objects()
     }
 
-    pub fn move_calls(&self) -> Vec<&MoveCall> {
+    pub fn legacy_move_calls(&self) -> Vec<&MoveCall> {
```

Comment:
Can we add a document here explaining what's legacy about these move calls, and why you may not want to use this anymore, after programmable transactions takes over the world (i.e. what goes wrong in that case, if you keep using this function).